packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -1,95 +1,84 @@
+		if (isWorkersWithAssets(options)) {
+			if (options.assetsBindingName) {
+				// binding between User Worker and Asset Worker
+				bindings.push({
+					name: options.assetsBindingName,
+					service: { name: ASSETS_SERVICE_NAME },
+				});
+			}
+		}
+		return bindings;
+	},
 
-export function isWorkersWithAssets(
-	options: z.infer<typeof KVOptionsSchema>
-): options is AssetsOptions {
-	return options.assetsPath !== undefined;
-}
+	async getNodeBindings(options) {
+		const bindings: Record<string, symbol> = {};
 
-const SERVICE_NAMESPACE_ASSET = `${KV_PLUGIN_NAME}:asset`;
+		if (isWorkersWithAssets(options)) {
+			if (options.assetsBindingName) {
+				// binding between User Worker and Asset Worker
+				bindings[options.assetsBindingName] = kProxyNodeBinding;
+			}
+		}
+		return bindings;
+	},
 
-export const buildAssetsManifest = async (dir: string) => {
-	const manifest = await walk(dir);
-	const sortedAssetManifest = sortManifest(manifest);
-	const encodedAssetManifest = encodeManifest(sortedAssetManifest);
-	return encodedAssetManifest;
+	async getServices({ options }) {
+		const services: Service[] = [];
+		if (isWorkersWithAssets(options)) {
+			// Router Worker, Asset Worker, Fake KV Worker
+			services.push(...(await getAssetService(options)));
+		}
+		return services;
+	},
 };
 
-export async function getAssetsBindings(
-	options: AssetsOptions
-): Promise<Worker_Binding[]> {
-	const assetsBindings = getAssetsBindingsNames(
-		options?.assetsKVBindingName,
-		options?.assetsManifestBindingName
-	);
-
-	const assetsManifest = await buildAssetsManifest(options.assetsPath);
-	return [
-		{
-			// this is the binding to the KV namespace that the assets are in.
-			name: assetsBindings.ASSETS_KV_NAMESPACE,
-			kvNamespace: { name: SERVICE_NAMESPACE_ASSET },
-		},
-		{
-			// this is the binding to an ArrayBuffer containing the binary-encoded
-			// assets manifest.
-			name: assetsBindings.ASSETS_MANIFEST,
-			data: assetsManifest,
-		},
-	];
-}
-
-export async function getAssetsNodeBindings(
-	options: AssetsOptions
-): Promise<Record<string, unknown>> {
-	const assetsManifest = buildAssetsManifest(options.assetsPath);
-	const assetsBindings = getAssetsBindingsNames(
-		options?.assetsKVBindingName,
-		options?.assetsManifestBindingName
-	);
-
-	return {
-		[assetsBindings.ASSETS_KV_NAMESPACE]: kProxyNodeBinding,
-		[assetsBindings.ASSETS_MANIFEST]: assetsManifest,
-	};
-}
-
-export async function getAssetsServices(
+export async function getAssetService(
 	options: AssetsOptions
 ): Promise<Service[]> {
 	const assetsReverseMap = await createReverseMap(options.assetsPath);
 
-	const storageServiceName = `${SERVICE_NAMESPACE_ASSET}:storage`;
+	const storageServiceName = `${ASSETS_PLUGIN_NAME}:storage`;
 	const storageService: Service = {
 		name: storageServiceName,
 		disk: { path: options.assetsPath, writable: true },
 	};
 	const namespaceService: Service = {
-		name: SERVICE_NAMESPACE_ASSET,
+		name: FAKE_KV_SERVICE_NAME,
```

Comment:
same comment as [here](https://github.com/cloudflare/workers-sdk/pull/6564/commits/522bae675f9b31193de03c20cfa5fd24a8f74509#r1735684791)

Reply:
I agree. "fake" doesn't add value. Try to identify all these assets with their feature like "assets"